src/core/execute-serialize.c

```diff
@@ -1116,38 +1116,65 @@ static int exec_runtime_serialize(const ExecRuntime *rt, FILE *f, FDSet *fds) {
 
-                        a = fdset_put_dup(fds, rt->shared->netns_storage_socket[0]);
+                        if (serialize_fd_index)
+                                a = fdset_put_indexed(fds, rt->shared->netns_storage_socket[0]);
+                        else
+                                a = fdset_put_dup(fds, rt->shared->netns_storage_socket[0]);
                         if (a < 0)
                                 return a;
 
-                        b = fdset_put_dup(fds, rt->shared->netns_storage_socket[1]);
+                        if (serialize_fd_index)
+                                b = fdset_put_indexed(fds, rt->shared->netns_storage_socket[1]);
+                        else
+                                b = fdset_put_dup(fds, rt->shared->netns_storage_socket[1]);
                         if (b < 0)
                                 return b;
 
-                        r = serialize_item_format(f, "exec-runtime-netns-storage-socket", "%d %d", a, b);
+                        r = serialize_item_format(
+                                        f,
+                                        serialize_fd_index ? "exec-runtime-netns-storage-socket-by-fd-index" : "exec-runtime-netns-storage-socket",
+                                        "%d %d",
+                                        a,
+                                        b);
                         if (r < 0)
                                 return r;
                 }
 
                 if (rt->shared->ipcns_storage_socket[0] >= 0 && rt->shared->ipcns_storage_socket[1] >= 0) {
                         int a, b;
 
-                        a = fdset_put_dup(fds, rt->shared->ipcns_storage_socket[0]);
+                        if (serialize_fd_index)
+                                a = fdset_put_indexed(fds, rt->shared->ipcns_storage_socket[0]);
+                        else
+                                a = fdset_put_dup(fds, rt->shared->ipcns_storage_socket[0]);
                         if (a < 0)
                                 return a;
 
-                        b = fdset_put_dup(fds, rt->shared->ipcns_storage_socket[1]);
+                        if (serialize_fd_index)
+                                b = fdset_put_indexed(fds, rt->shared->ipcns_storage_socket[1]);
+                        else
+                                b = fdset_put_dup(fds, rt->shared->ipcns_storage_socket[1]);
                         if (b < 0)
                                 return b;
 
-                        r = serialize_item_format(f, "exec-runtime-ipcns-storage-socket", "%d %d", a, b);
+                        r = serialize_item_format(
+                                        f,
+                                        serialize_fd_index ? "exec-runtime-ipcns-storage-socket-by-fd-index" : "exec-runtime-ipcns-storage-socket",
+                                        "%d %d",
+                                        a,
+                                        b);
```

Comment:
sorry, but please add a common implementatin for this, it's not ok to copy this super complex code all over the place, with just a different prefix